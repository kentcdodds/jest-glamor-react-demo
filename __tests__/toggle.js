import React from 'react'
import {render, mount} from 'enzyme'
import Toggle from '../toggle'

test('has off styles applied by default', () => {
  const wrapper = renderToggle()
  expect(wrapper).toMatchSnapshot()
})

test('has on styles applied when initialToggledOn: true', () => {
  const wrapper = renderToggle({initialToggledOn: true})
  expect(wrapper).toMatchSnapshot()
})

test('invokes the onToggle prop when clicked', () => {
  const onToggle = jest.fn()
  const wrapper = mountToggle({onToggle})
  clickButton(wrapper)
  expect(onToggle).toHaveBeenCalledTimes(1)
  expect(onToggle).toBeCalledWith(true)
})

/**
 * Uses enzyme to mount the Toggle component
 * @param {Object} props - the props to mount the component with
 * @return {Object} - the enzyme wrapper
 */
function mountToggle(props = {}) {
  return mount(<Toggle onToggle={() => {}} children="Toggle Me" {...props} />)
}

/**
 * Uses enzyme to render the Toggle component
 * @param {Object} props - the props to render the component with
 * @return {Object} - the enzyme wrapper
 */
function renderToggle(props = {}) {
  return render(<Toggle onToggle={() => {}} children="Toggle Me" {...props} />)
}

/**
 * finds the button in the given wrapper and simulates a click event
 * @param {Object} wrapper - the enzyme wrapper
 */
function clickButton(wrapper) {
  wrapper.find(sel('button')).simulate('click')
}

// this helper will make it easier for you to find
// labeled elements in the wrapper:
// const tagInput = wrapper.find(sel('tags'))
function sel(id) {
  return `[data-test="${id}"]`
}
