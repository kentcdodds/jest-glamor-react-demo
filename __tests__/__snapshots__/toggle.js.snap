// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has off styles applied by default 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline-block;
  padding: 6px 12px;
  margin-bottom: 0px;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.4;
  text-align: center;
  cursor: pointer;
  border-radius: 4px;
  color: #fff;
  background-color: #337ab7;
  border-color: #285f8f;
}

.glamor-0:hover,
[data-glamor-0]:hover,
.glamor-0:active,
[data-glamor-0]:active,
.glamor-0:focus,
[data-glamor-0]:focus {
  background-color: #285f8f;
  border-color: #1d4567;
}

<button
  class="glamor-0"
  data-test="button"
>
  Toggle Me
</button>
`;

exports[`has on styles applied when initialToggledOn: true 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline-block;
  padding: 6px 12px;
  margin-bottom: 0px;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.4;
  text-align: center;
  cursor: pointer;
  border-radius: 4px;
  color: #fff;
  background-color: #22527b;
  border-color: #173853;
}

.glamor-0:hover,
[data-glamor-0]:hover,
.glamor-0:active,
[data-glamor-0]:active,
.glamor-0:focus,
[data-glamor-0]:focus {
  background-color: #1d4567;
  border-color: #122a3f;
}

<button
  class="glamor-0"
  data-test="button"
>
  Toggle Me
</button>
`;
